src/main/java/edu/kit/hci/soli/controller/EventFeedController.java

```diff
@@ -0,0 +1,42 @@
+package edu.kit.hci.soli.controller;
+
+import edu.kit.hci.soli.service.BookingsService;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.ResponseEntity;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@RestController("/api/events")
+public class EventFeedController {
+    final BookingsService bookingsRepository;
+
+    public EventFeedController(BookingsService bookingsRepository) {
+        this.bookingsRepository = bookingsRepository;
+    }
+
+    // https://fullcalendar.io/docs/events-json-feed
+    @GetMapping("/api/events")
+    @Transactional(readOnly = true)
+    public List<BookingsService.CalendarEvent> getEvents(
+            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime start,
+            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime end
+    ) {
+        //TODO we should highlight events by ourselves WITHOUT exposing the ownership by allowing others to send requests as spoofed users
+        if (end.isBefore(start)) {
+            throw new IllegalArgumentException("End date must be after start date");
+        }
+        if (end.minusMonths(3).isAfter(start)) {
+            throw new IllegalArgumentException("Time range must be less than 3 months");
+        }
```

Comment:
This is to prevent a DOS attack requesting all events after the system was in use a while.
Naturally, this restriction could be changed and others could be added if so desired.